### PR TITLE
ENH: Enable resampling morphometrics to fsLR CIFTI-2 files

### PR DESCRIPTION
**Pull request by effigies**
Currently inaccessible to smriprep itself because I don't want to add and test a CLI option. Could probably go into a bug-fix release if someone cares about it before the next minor release.

### CODE REVIEW

**Comment by effigies** [comment]
@mgxd I would appreciate a review here. I'll fix the flake8 failure (unused import) just before merging. No need to cycle the CI for it.

File: smriprep/workflows/surfaces.py [hunk 1]
```diff
@@ -823,6 +825,195 @@ def init_anat_ribbon_wf(name="anat_ribbon_wf"):
...
+        List of BOLD files resampled on the fsaverage (ico7) surfaces.
+    surf_refs :
+        List of unique identifiers corresponding to the BOLD surface-conversions.
+
+    Outputs
+    -------
+    cifti_bold : :obj:`str`
+        BOLD CIFTI dtseries.
+    cifti_metadata : :obj:`str`
+        BIDS metadata file corresponding to ``cifti_bold``.
```
Comment (mgxd) [inline smriprep/workflows/surfaces.py:866]: These should reflect the input/output nodes

**Review by mgxd — COMMENTED** [review]
Nice!

Very minor comments, otherwise looks good to me. It would be good to have @madisoth (or someone else working with CIFTI outputs) verify these outputs are similar to those generated from the `wb_command`s, to ensure we can be as 1:1 as possible.

**Comment by effigies** [comment]
@mgxd @madisoth Just sneaking the recon-all fix in here, rather than making a new PR.

**Comment by effigies** [comment]
@madisoth: You're probably running into this bug:

```
230303-14:00:55,586 nipype.workflow INFO:                                                                                                                                                                                                                     
         [Node] Executing "select_midthick" <nipype.interfaces.utility.base.Select>                                                                                                                                                                           
230303-14:00:55,588 nipype.workflow INFO:                                                                                                                                                                                                                     
         [Node] Finished "select_midthick", elapsed time 0.000265s.                                                                                                                                                                                           
230303-14:00:55,588 nipype.workflow WARNING:                                                                                   
         Storing result file without outputs                                                                                                                                                                                                                  
230303-14:00:55,588 nipype.workflow WARNING:                                                                                                                                                                                                                  
         [Node] Error on "fmriprep_23_0_wf.single_subject_01_wf.anat_preproc_wf.anat_ribbon_wf.select_midthick" (/scratch/fmriprep_23_0_wf/single_subject_01_wf/anat_preproc_wf/anat_ribbon_wf/select_midthick)                                               
230303-14:00:55,590 nipype.workflow ERROR:                                                                                                                                                                                                                    
         Node select_midthick failed to run on host 0e2f22be8776.                                                                                                                                                                                             
230303-14:00:55,590 nipype.workflow ERROR:                                                                                                                                                                                                                    
         Saving crash info to /out/sub-01/log/20230303-133746_ec8b0c21-aeca-4b28-acf1-5262714e1640/crash-20230303-140055-root-select_midthick-c36f3596-25d1-4908-95ef-76b05b799d1d.txt                                                                        
Traceback (most recent call last):                                                                                                                                                                                                                            
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/plugins/multiproc.py", line 344, in _send_procs_to_workers                                                                                                                                     
    self.procs[jobid].run(updatehash=updatehash)                                                                                                                                                                                                              
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 527, in run                                                                                                                                                             
    result = self._run_interface(execute=True)                                                                                                                                                                                                                
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 645, in _run_interface                                                                                                                                                  
    return self._run_command(execute)                                                                                          
  File "/opt/conda/lib/python3.9/site-packages/nipype/pipeline/engine/nodes.py", line 771, in _run_command                                                                                                                                                    
    raise NodeExecutionError(msg)                                                                                                                                                                                                                             
nipype.pipeline.engine.nodes.NodeExecutionError: Exception raised while executing Node select_midthick.                                                                                                                                                       
                                                                                                                                                                                                                                                              
Traceback:                                                                                                                                                                                                                                                    
        Traceback (most recent call last):                                                                                                                                                                                                                    
          File "/opt/conda/lib/python3.9/site-packages/nipype/interfaces/base/core.py", line 401, in run                                                                                                                                                      
            outputs = self.aggregate_outputs(runtime)                                                                          
          File "/opt/conda/lib/python3.9/site-packages/nipype/interfaces/base/core.py", line 430, in aggregate_outputs                                                                                                                                        
            predicted_outputs = self._list_outputs()  # Predictions from _list_outputs                                                                                                                                                                        
          File "/opt/conda/lib/python3.9/site-packages/nipype/interfaces/utility/base.py", line 412, in _list_outputs                                                                                                                                         
            out = np.array(self.inputs.inlist, dtype=object)[                                                                                                                                                                                                 
        IndexError: index 6 is out of bounds for axis 0 with size 6  
```

The latest commit fixes that.

**Comment by effigies** [comment]
Just a note that I've now run through a downsampled test dataset successfully including this and the three open PRs on fMRIPrep. Rerunning now with full res ds005/sub-01 from scratch.

**Pull request merged**